corehq/apps/app_manager/xform.py

```diff
@@ -1014,22 +1019,23 @@ def get_questions(self, langs, include_triggers=False,
             questions.append(question)
 
         repeat_contexts = sorted(repeat_contexts, reverse=True)
+        group_contexts = sorted(group_contexts, reverse=True)
 
         save_to_case_nodes = {}
         for path, data_node in six.iteritems(leaf_data_nodes):
             if path not in excluded_paths:
                 bind = self.get_bind(path)
-                try:
-                    matching_repeat_context = [
-                        rc for rc in repeat_contexts if path.startswith(rc + '/')
-                    ][0]
-                except IndexError:
-                    matching_repeat_context = None
+
+                matching_repeat_context = (
+                    [rc for rc in repeat_contexts if path.startswith(rc + '/')] + [None])[0]
```

Comment:
The `+ [None])[0]` construct would be confusing if I couldn't see the try/except block it's replacing. I think the original was easier to read.